lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -10033,6 +10032,24 @@ namespace Js
         return true;
     }
 
+    bool JavascriptOperators::CheckIfPrototypeChainContainsProxyObject(RecyclableObject* prototype)
+    {
+        Assert(JavascriptOperators::IsObjectOrNull(prototype));
+
+        if (prototype->GetTypeId() == TypeIds_Null)
+        {
+            return false;
+        }
+        else if (prototype->GetTypeId() == TypeIds_Proxy)
+        {
+            return true;
+        }
+        else
+        {
+            return CheckIfPrototypeChainContainsProxyObject(prototype->GetPrototype());
```

Comment:
Even though prototype chains are typically fairly short, I think it is probably a good idea to convert this function to an interative version to avoid taking up O(n) stack frames where n is the length of the prototype chain.